src/sidebar/oauth-auth.js

```diff
@@ -104,56 +153,108 @@ function auth($http, $window, flash, random, settings) {
     });
   }
 
-  // Exchange the refresh token for a new access token and refresh token pair.
-  // See https://tools.ietf.org/html/rfc6749#section-6
-  function refreshAccessToken(refreshToken) {
-    var data = {grant_type: 'refresh_token', refresh_token: refreshToken};
-    postToTokenUrl(data).then(function (response) {
+  /**
+   * Exchange the refresh token for a new access token and refresh token pair.
+   * See https://tools.ietf.org/html/rfc6749#section-6
+   *
+   * @param {string} refreshToken
+   * @param {RefreshOptions} options
+   * @return {Promise<string|null>} Promise for the new access token
+   */
+  function refreshAccessToken(refreshToken, options) {
+    var data = { grant_type: 'refresh_token', refresh_token: refreshToken };
+    return postToTokenUrl(data).then((response) => {
       var tokenInfo = tokenInfoFrom(response);
-      refreshAccessTokenBeforeItExpires(tokenInfo);
+
+      if (options.persist) {
+        saveToken(tokenInfo);
+      }
+
+      refreshAccessTokenBeforeItExpires(tokenInfo, {
+        persist: options.persist,
```

Comment:
The reason for passing an `options` dict here is because JS doesn't have keyword args and I wanted the calls to this function to be easier to read. Compare:

```
refreshAccessToken(tokenInfo, true)
```

vs.

```
refreshAccessToken(tokenInfo, { persist: true })
```